pkg/mount/dag_steps.go

```diff
@@ -715,3 +715,42 @@ func (s *State) UKIUnlock(g *herd.Graph, opts ...herd.OpOption) error {
 		return kcrypt.UnlockAll(true)
 	}))...)
 }
+
+// MountLiveCd tries to mount the livecd if we are booting from one into /run/initramfs/live
+// to mimic the same behavior as the livecd on non-uki boot.
+func (s *State) MountLiveCd(g *herd.Graph, opts ...herd.OpOption) error {
+	return g.Add(cnst.OpUkiMountLivecd, append(opts, herd.WithCallback(func(ctx context.Context) error {
+		// If we are booting from Install Media
+		if !internalUtils.EfiBootFromInstall() {
```

Comment:
minor request but I personally would prefer an early return, so the rest of the code is not nested even deeper

Reply:
This is what you mentioned or did I got this totally incorrect?